src/inventory.h

```diff
@@ -122,20 +122,13 @@ struct ItemStack
 	// Returns true if the item is (was) a tool
 	bool addWear(s32 amount, IItemDefManager *itemdef)
 	{
-		if(getDefinition(itemdef).type == ITEM_TOOL)
-		{
-			if(amount > 65535 - wear)
-				clear();
-			else if(amount < -wear)
-				wear = 0;
-			else
-				wear += amount;
-			return true;
-		}
+		if(amount > 65535 - wear && getDefinition(itemdef).type == ITEM_TOOL)
```

Comment:
The constant would be called `U16_MAX`. There can be no higher wear than that, so it might be a good idea to replace this now.